File: tree/heightoftree.cpp
class Solution{
    public:
    //Function to find the height of a binary tree.
    int height(struct Node* node){
    if(node==NULL){
        return 0;
    }
    int leftsubtree=height(node->left);//heigtht of left subtree.
    int rightsubtree=height(node->right);//height of right subtree.
    
    int ans=max(leftsubtree,rightsubtree)+1;//store the max of left and right subtree and add 1 for the root node.
    
    return ans;
    }

File: tree/heightoftree.js
//class Node{
   constructor(val){
this.val=val;
this.left=null;
this.right=null;
   }
}//


var maxDepth = function(root) {
    if (root===null){
        return 0;
    }
    var leftSubtree= maxDepth(root.left);
    var rightSubtree =maxDepth(root.right);

    var result = Math.max(leftSubtree,rightSubtree)+1;

    return result;
};
